pkg/api/types.go

```diff
@@ -2799,6 +2799,16 @@ type NodeSpec struct {
 	// If specified, the node's taints.
 	// +optional
 	Taints []Taint
+
+	// If specified, the source to get node configuration from
+	// The DynamicKubeletConfig feature gate must be enabled for the Kubelet to use this field
+	// +optional
+	ConfigSource *NodeConfigSource
```

Comment:
Is this preferred over gating it in `pkg/api/validation/validation.go` or should I do both?